### ISSUE TITLE
Support for RFC9606 RRType 261 (RESINFO) 

### ISSUE THREAD

**Issue by mdavids** [PowerDNS/pdns#14983]
- Program: dnsdist
 - Issue type: Feature request

### Short description
RFC9606 introduces a new RRtype 'RESINFO'. It would be nice if that could be used with 'DNSQType'.

### Usecase
Implement RFC9606 support for our public resolver. The RFC says: "DNS client MUST set the Recursion Desired (RD) bit of the query to 0. The DNS client MUST discard the response if the AA flag in the response is set to 0", which implicates that just adding RESINFO-records to a zone file is not the way to go (see https://marc.info/?l=openbsd-tech&m=173368282507730&w=2).

With a Luarule this could however be done, when the RRType is supported in the 'DNSQtype'-function. Looking at the source code, this is [not the case yet](https://github.com/PowerDNS/pdns/blob/fe75d35ed456cc8af18e078de211002388689241/pdns/qtype.cc#L33).

### Description
Simply extend the list of RRtypes with RESINFO.

**Comment by rgacogne** [comment]
Makes sense to me! Note that you should be able to use the numerical value in most places in the meantime.

**Comment by mdavids** [comment]
> Makes sense to me! Note that you should be able to use the numerical value in most places in the meantime.

Please enlighten me; how would that work? It's not `DNSQType.261`, is it?

**Comment by omoerbeek** [comment]
[QTypeRule(qtype)[¶](https://dnsdist.org/reference/selectors.html?highlight=qtyperule#QTypeRule)](https://dnsdist.org/reference/selectors.html?highlight=qtyperule#QTypeRule) says it taks an int, so e.g. `QTypeRule(261)` should do it.
